types/field_type.go

```diff
@@ -146,6 +147,35 @@ func hasVariantFieldLength(tp *FieldType) bool {
 	return false
 }
 
+// http://cavaliercoder.com/blog/optimized-abs-for-int64-in-go.html
+func abs(n int64) int64 {
+	y := n >> 63       // y ← x >> 63
+	return (n ^ y) - y // (x ⨁ y) - y
+}
+
+var uintSizeTable = [20]uint64{
+	9, 99, 999, 9999, 99999,
+	999999, 9999999, 99999999, 999999999, 9999999999,
+	99999999999, 999999999999, 9999999999999, 99999999999999, 999999999999999,
+	9999999999999999, 99999999999999999, 999999999999999999, 9999999999999999999,
+	math.MaxUint64} // 18446744073709551615 20 digits in total
+
+func lenOfUint64Fast(x uint64) int {
+	for i := 0; ; i++ {
+		if x <= uintSizeTable[i] {
+			return i + 1
```

Comment:
Maybe we can use a redundant data in uintSizeTable[0] so that we can return `i` instead of `i+1`?

Reply:
You are smart :) One operation less, but it slightly increases the difficulty to understand the code perhaps :) I will change it and you give it a look :D

Reply:
I think it's OK :)